gapidapk/android/app/src/main/java/com/google/android/gapid/PackageInfoService.java

```diff
@@ -314,37 +339,47 @@ private JSONObject getPackageJson(
-            }
-
-            Drawable drawable = null;
             try {
-                drawable = resources.getDrawableForDensity(iconId, iconDensity);
-            } catch (Resources.NotFoundException ex) {
-                return -1;
-            }
-            if (drawable == null || !(drawable instanceof BitmapDrawable)) {
-                return -1;
-            }
+                if (resources == null || iconId <= 0) {
+                    return -1;
+                }
 
-            Bitmap bitmap = ((BitmapDrawable) drawable).getBitmap();
-            if (mBitmapMap.containsKey(bitmap)) {
-                return mBitmapMap.get(bitmap);
-            }
+                Drawable drawable;
+                try (Counter.Scope t = Counter.time("getDrawableForDensity")){
+                    drawable = resources.getDrawableForDensity(iconId, iconDensity);
+                } catch (Resources.NotFoundException ex) {
+                    return -1;
+                }
+                if (drawable == null || !(drawable instanceof BitmapDrawable)) {
+                    return -1;
+                }
+
+                Bitmap bitmap = ((BitmapDrawable) drawable).getBitmap();
+                if (mBitmapMap.containsKey(bitmap)) {
+                    return mBitmapMap.get(bitmap);
+                }
 
-            ByteArrayOutputStream stream = new ByteArrayOutputStream();
-            bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
-            byte[] pngBytes = stream.toByteArray();
-            String pngBase64 = Base64.encodeToString(pngBytes, Base64.NO_WRAP);
-            if (!mMap.containsKey(pngBase64)) {
-                int index = mJson.length();
-                mMap.put(pngBase64, index);
-                mBitmapMap.put(bitmap, index);
-                mJson.put(pngBase64);
-                return index;
-            } else {
-                return mMap.get(pngBase64);
+                ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                try (Counter.Scope t1 = Counter.time("bitmap.compress")) {
+                    try (Counter.Scope t2 = Counter.time(bitmap.getWidth() + "x" + bitmap.getHeight())) {
+                        bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+                    }
+                }
+                byte[] pngBytes = stream.toByteArray();
+                String pngBase64;
+                try (Counter.Scope t = Counter.time("Base64.encodeToString")) {
+                    pngBase64 = Base64.encodeToString(pngBytes, Base64.NO_WRAP);
+                }
+                if (!mMap.containsKey(pngBase64)) {
+                    int index = mJson.length();
+                    mMap.put(pngBase64, index);
+                    mBitmapMap.put(bitmap, index);
+                    mJson.put(pngBase64);
+                    return index;
+                } else {
+                    return mMap.get(pngBase64);
+                }
+            } finally {
```

Comment:
You are undoing most of this change in the next commit. Maybe squash the 'add try, remove try' parts?